src/03-ticket-calculator.js

```diff
@@ -109,7 +122,41 @@ function calculateTicketPrice(ticketData, ticketInfo) {}
     purchaseTickets(tickets, purchases);
     //> "Ticket type 'discount' cannot be found."
  */
-function purchaseTickets(ticketData, purchases) {}
+function purchaseTickets(ticketData, purchases) {
+  let result = ""; //Holds answer
+  let totalCost = 0; //Holds total cost
+  let personCost = 0; //Holds the cost of each person
+  let entrantType = "" //Holds the entrat type
+  let extras = "" //holds the description 
+ 
+ for (let i = 0; i < purchases.length; i++){ //Loops through purchases array
+  if (ticketData[purchases[i].ticketType] === undefined){ //Checks if ticket type of ticketData is undefined 
+    return "Ticket type 'incorrect-type' cannot be found."; //Returns the error message for the incorrect ticket type
+    } else if (ticketData[purchases[i].ticketType].priceInCents[purchases[i].entrantType] === undefined){ //Checks if entrant type of ticketData is undefined
+      return "Entrant type 'incorrect-entrant' cannot be found."; //Returns the error message for the incorrect entrant type
+    } else if (purchases[i].extras[0] === "incorrect-extra"){ //checks if the given extra is equal to "incorrect-extra"
+      return "Extra type 'incorrect-extra' cannot be found."; //Returns the error message for the incorrect extra
+    }
```

Comment:
Same note as above here. Your returns can be more dynamic. Also another approach you could have taken was to leverage the `calculateTicketPrice` function. If the return is a string then your know it is an error message and return that, if not you would add the price for it ticket to a total .